### PR TITLE
[MPS] Restrict MSELoss to floating types

### PR DESCRIPTION
**Pull request by malfet**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #139960
* #139959
* #139961

Becuase if invoked with long type it crahses deep in MPSGraph framework and to keep parity with CPU

Add test that validates that if dtype is not floating, both CPU and MPS implementations will error out
Fix function name for `mse_loss_out_mps` as `__func__` for any structured op implementation is `impl`

Fixes https://github.com/pytorch/pytorch/issues/139723

### CODE REVIEW

File: aten/src/ATen/native/mps/operations/LossOps.mm [hunk 1]
```diff
@@ -995,15 +995,17 @@ Tensor huber_loss_mps(const Tensor& input, const Tensor& target, int64_t reducti
 
 // MSELoss
 TORCH_IMPL_FUNC(mse_loss_out_mps)(const Tensor& input, const Tensor& target, int64_t reduction, const Tensor& output_) {
-  string op_name = __func__;
```
Comment (kimishpatel) [inline aten/src/ATen/native/mps/operations/LossOps.mm:998]: why this change?

**Review by kimishpatel — APPROVED** [review]

File: aten/src/ATen/native/mps/operations/LossOps.mm (hunk 1 above)
Comment (malfet) [reply aten/src/ATen/native/mps/operations/LossOps.mm:998]: See PR description, for `TORCH_IMPL_FUNC` macro `__func__` will always return `impl`, which is bad for MPSGraph cache keys as well as for error messages.
